optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/forager/finalist/StrategicOscillationByLevelFinalistPodium.java

```diff
@@ -59,45 +63,77 @@ public void addMove(LocalSearchMoveScope moveScope) {
             finalistScore = null;
             finalistLevelNumbers = null;
         }
-        Score moveScore = moveScope.getScore();
-        Number[] moveLevelNumbers = moveScore.toLevelNumbers();
-        int comparison = doComparison(moveScore, moveLevelNumbers);
+        int comparison = doComparison(moveScope);
         if (comparison > 0) {
-            finalistScore = moveScore;
-            finalistLevelNumbers = moveLevelNumbers;
+            finalistScore = moveScope.getScore();
+            finalistLevelNumbers = moveScope.getScore().toLevelNumbers();
             finalistList.clear();
             finalistList.add(moveScope);
         } else if (comparison == 0) {
             finalistList.add(moveScope);
         }
     }
 
-    private int doComparison(Score moveScore, Number[] moveLevelNumbers) {
+    private int doComparison(LocalSearchMoveScope moveScope) {
         if (finalistScore == null) {
             return 1;
         }
-        for (int i = 0; i < referenceLevelNumbers.length; i++) {
-            boolean moveIsHigher = ((Comparable) moveLevelNumbers[i]).compareTo(referenceLevelNumbers[i]) > 0;
-            boolean finalistIsHigher = ((Comparable) finalistLevelNumbers[i]).compareTo(referenceLevelNumbers[i]) > 0;
-            if (moveIsHigher) {
-                if (finalistIsHigher) {
-                    break;
-                } else {
-                    return 1;
+
+        Score moveScore = moveScope.getScore();
+        Score bestStepScore = moveScope.getStepScope().getScore();
```

Comment:
This is always null at this point. The StepScope.getScore() gets filled in after the forager selects a winning move (which it gets from this podium).